pkg/util/mount/mount_linux.go

```diff
@@ -318,10 +318,9 @@ func getDeviceNameFromMount(mounter Interface, mountPath, pluginDir string) (str
 		klog.V(4).Infof("Directory %s is not mounted", mountPath)
 		return "", fmt.Errorf("directory %s is not mounted", mountPath)
 	}
-	basemountPath := path.Join(pluginDir, MountsInGlobalPDPath)
 	for _, ref := range refs {
-		if strings.HasPrefix(ref, basemountPath) {
-			volumeID, err := filepath.Rel(basemountPath, ref)
+		if strings.HasPrefix(ref, pluginDir) {
+			volumeID, err := filepath.Rel(pluginDir, ref)
```

Comment:
Seems like `pluginDir`, that comes from the plugin's `ConstructVolumeSpec`, is no longer only the plugin directory. Now it has the `MountsInGlobalPDPath` const appended to it, right?

We might have to change the `mount.Interface` as well.

Reply:
That was what I was going for here, yes. I was trying to make as few changes as possible, without having to change `mount.Interface`.

An alternative would be to go ahead and change the interface from:
```go
GetDeviceNameFromMount(mountPath, pluginDir string)
```
to
```go
GetDeviceNameFromMount(mountPath, pluginDir, mountDir string)
```
And then append the mount dir in the function, which is what it did before.

The overall goal here is not have anything in `pkg/util/mount` depend on access to that constant, which needs to live in another package (I had put it in `pkg/volume/util`).

I'm fine with either method.

Reply:
I prefer changing it to the first

Reply:
Sorry I wasn't very clear.  I preferred only having two arguments to `GetDeviceNameFromMount`, and having some helper method that appends the constant to the plugin dir.

Reply:
doh. okay, I can do that. 😆